src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private Object[] array = new Object[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        if (size == array.length) {
+            array = Arrays.copyOf(array, (int) (array.length * GROWTH_FACTOR));
+        }
```

Comment:
```suggestion
        if (size == array.length) {
            array = Arrays.copyOf(array, (int) (array.length * GROWTH_FACTOR));
        }
```

This logic occurs in several methods. Move it to a separate private method.